### PR TITLE
Add environment parsing support for enums.

### PR DESCRIPTION
**Pull request by kschwab**
Adds support for environment parsing of enum field names to values.

### CODE REVIEW

File: tests/test_settings.py [hunk 1]
```diff
@@ -1853,6 +1854,33 @@ class Settings(BaseSettings):
...
+        kiwi = 1
+        lime = 2
+
+    class Settings(BaseSettings):
+        fruit: FruitsEnum
+
+    with pytest.raises(ValidationError):
+        env.set('FRUIT', 'kiwi')
+        s = Settings()
+        assert s.fruit == FruitsEnum.kiwi
```
Comment (hramezani) [inline tests/test_settings.py:1869]: I think this assertion is not needed here

**Comment by hramezani** [comment]
Thanks @kschwab for the new feature implementation.

I left some suggestion

**Comment by hramezani** [comment]
please update

File: tests/test_settings.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (kschwab) [reply tests/test_settings.py:1869]: This is much better; I had not thought of this. Thanks @hramezani, will validate exceptions in future.

**Review by hramezani — APPROVED** [review]

**Pull request merged**